chanzuckerberg/idseq-web
# Add custom boolean toggle control for water control metadata

# Description

Following the mocks https://chanzuckerberg.invisionapp.com/share/A6V572ZSBU5#/screens/396544827/comments, this adds a custom control based on semantic UI `Checkbox`. Rendering the control is determined by the new `boolean?` method of `MetadataField`. Currently, only fields that have options "Yes" and "No" are considered boolean.

![image](https://user-images.githubusercontent.com/28797/71937097-2e33f580-3160-11ea-95fd-eff2a240b604.png)

![image](https://user-images.githubusercontent.com/28797/71860117-80184500-30a6-11ea-9928-e059b24f8574.png)

I also added an ordering of fields based on the design. 

# Notes

* please ignore the snyk warning because I'm going to merge this branch into another branch, not master
* there was a semantic ui CSS bug such that the default `Checkbox` with toggle is not rendering the circle when unselected
* `force_options` does not seem to do anything in the upload flow. All fields that have options are "forced" by the UI controls.
* only other yes/no field I saw in the codebase was "antibiotics applied"
* I refactored the playground to use `children` as that is more idiomatic

# Tests

* new rspec test of `boolean?`
* load playground, see both states of toggle, see click behavior
* load upload flow, check apply all works
